### PR TITLE
Banned routes generator

### PR DESCRIPTION
**Pull request by amy-corson-ibigroup**
In cases where there are route mode overrides in the config, we'll use the banned parameter to exclude routes under a custom submode. This adds a util function to generate a list of banned routes based on the mode settings and route mode overrides.

### CODE REVIEW

File: packages/trip-form/src/MetroModeSelector/utils.ts [hunk 1]
```diff
@@ -25,6 +25,33 @@ export function aggregateModes(
...
+ * when their mode is deselected.
+ */
+export function getBannedRoutesFromSubmodes(
+  modeSettings: ModeSetting[],
+  routeModeOverrides: Record<string, string>
+): string[] {
+  const disabledSubmodes = modeSettings.reduce<string[]>((prev, cur) => {
+    if (cur.type === "SUBMODE") {
+      if (!cur.value) {
+        const modeName = cur.modeOverride || cur.addTransportMode.mode;
```
[the remark this answers is not in the document]
Comment (daniel-heppner-ibigroup) [reply packages/trip-form/src/MetroModeSelector/utils.ts:41]: I don't think this property exists, please remove it. Only check for `addTransportMode.mode`.

**Review by daniel-heppner-ibigroup — CHANGES_REQUESTED** [review]
A few things that need to be fixed here

**Review by daniel-heppner-ibigroup — APPROVED** [review]
I've changed this around to how I think it should work. Let me know what you think! 
The reason I prefer this is because it doesn't change the meaning of any existing properties.

**Comment by amy-corson-ibigroup** [comment]
@daniel-heppner-ibigroup Makes sense & much cleaner, thank you!!

**Review by miles-grant-ibigroup — APPROVED** [review]
This is such a clean solution for such a wild feature! Great work

**Pull request merged**